src/models/SearchQuery.js

```diff
@@ -70,15 +70,15 @@ export default class SearchQuery {
     return null;
   }
 
-  addFilter(filter) {
+  addFilter(filter, skipMerge) {
     const filterized = filterize(filter);
     const hash = filterized.getHash();
     // check if the filter do not exists.
     if (this.filtersIds.indexOf(hash) === -1) {
       // if it does not exist, check if a filter with the same type and context exists.
       const similarFilterIdx = this.filters.findIndex(d => d.context === filterized.context && d.type === filterized.type);
```

Comment:
can we add the op? I think it will be useful not to merge AND and OR filters of the same type

Reply:
I do not understand how adding an operator would fix the issue with regard to navigating from I&C page to search with two filters of the same type (say `person`), same op (`OR`) and with a single ID in the `q` (say `['id-of-albert-einstein']`). Maybe you can add this code if you understand it?

Reply:
think more in terms of solr query,
e.g; we already have an already existing filter `type=person` and op=`OR`, that translates to `q=(ent:eintsein OR ent:bohr)` 
if we add another filter, using the AND operator we don't want to merge them, we want to have:
q=(ent:eintsein OR ent:bohr) AND (ent:hoppenheimer AND ent:bohr)`

Reply:
moreover, if we add another entity `"new ent"` to the ispect and compare, and we just add the AND operator, the filters will merge into:
`q=(ent:eintsein OR ent:bohr) AND (ent:hoppenheimer AND ent:bohr AND ent:"new ent")`

Reply:
That's correct. Although it is not the case of the issue that this PR was trying to solve. In the long run we probably want a to have a more comprehensive function that takes a list of filters and optimises them by merging those of the same type that can be merged. In particular it would merge a list of filters like this:

```json
[
{
  "type": "person",
  "op": "OR",
  "q": ["einstein"]
},
{
  "type": "person",
  "op": "OR",
  "q": ["bohr"]
}
]
```

into 

```json
[
{
  "type": "person",
  "op": "AND",
  "q": ["einstein", "bohr"]
}
]
```

Then this function can be applied only in places where it would not confuse the user.

Reply:
I now understand better the context, but then woudn't  be better to do this magic merge only in the button? because in the normal search, we could enrich each filter, like this
```[
{
  "type": "person",
  "op": "OR",
  "q": ["einstein", "new ent"]
},
{
  "type": "person",
  "op": "OR",
  "q": ["bohr"]
}
]```
correctly translated to `(ent:einstein OR ent: "new ent") AND (ent:bohr)`